database/migrations/default/1744126149550_noncr3_soft_del_dedupe/up.sql

```diff
@@ -0,0 +1,56 @@
+--
+-- add fk column to atd_apd_blueform.case_id
+--
+alter table ems__incidents add column atd_apd_blueform_case_id integer;
+alter table ems__incidents
+add constraint ems__incidents_atd_apd_blueform_case_id_fkey foreign key (
+    atd_apd_blueform_case_id
+)
+references public.atd_apd_blueform (
+    case_id
+) on update cascade on delete set null;
+
+comment on column ems__incidents.atd_apd_blueform_case_id is 'The non-CR3 case ID matched to this record';
+
+--
+-- add soft delete column
+--
+alter table atd_apd_blueform add is_deleted boolean not null default false;
+
+comment on column atd_apd_blueform.is_deleted is 'Indicates soft-deletion';
+
+--
+-- add some indexes
+--
+create index ems__incidents_atd_apd_blueform_case_id_index on public.ems__incidents (
+    atd_apd_blueform_case_id
+);
+
+create index atd_apd_blueform_is_deleted_index on public.atd_apd_blueform (
+    is_deleted
+);
+
+create index crashes_case_id_idx on public.crashes (
+    case_id
+);
+
+--
+-- Trigger which soft-deletes atd_apd_blueform records if there is a cr3 crash
+-- with the same case_id
+--
+create or replace function remove_dupe_non_cr3s()
+returns trigger as $$
+DECLARE
+    text_case_id text;
+BEGIN
+    text_case_id := NEW.case_id::text;
+    if exists (select 1 from crashes where case_id = text_case_id) then
```

Comment:
afaik `case_id` on the `crashes` table is not unique, so how can we trust them to have 1:1 relationships with the non cr3 crashes?

Reply:
i guess its been some time since ive had context on `crash.case_id`, i thought i remembered that there are sometimes diff crashes with the same `case_id` so they cant be relied on to identify a crash, so it seemed weird to me that we are using them to identify and soft delete a non cr3. but maybe im lost on what the intent is here. 

thats an interesting idea to add `investigat_agency_id` to the mix, i just ran a query to see if we ever have duplicate `crash.case_id`'s where the agencies are the same, and my query returned around 200 instances of crashes w the same case id and investigat agencies. some of them seem to be complete duplicates, while others seem like they are two distinct crashes. :thinking: